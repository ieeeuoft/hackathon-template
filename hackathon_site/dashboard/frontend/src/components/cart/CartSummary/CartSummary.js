import React from "react";
import styles from "./CartSummary.module.scss";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TitledPaper from "components/general/TitledPaper/TitledPaper";

const CartSummary = ({ cartQuantity }) => (
    <TitledPaper title="Cart Summary">
        <Container className={styles.qty}>
            <Typography variant="body2">Quantity</Typography>
            <Typography variant="body2">{cartQuantity}</Typography>
        </Container>
        <Typography variant="body2" className={styles.msg}>
            Your entire team's order is here. Before you submit, refresh the page if
            someone added something new.
        </Typography>
        <Button
            color="primary"
            variant="contained"
            className={styles.btn}
            onClick={() => {
                alert("this would submit the order and redirect to dashboard");
            }}
            disableElevation
        >
            Submit order
        </Button>
    </TitledPaper>
);

export default CartSummary;
