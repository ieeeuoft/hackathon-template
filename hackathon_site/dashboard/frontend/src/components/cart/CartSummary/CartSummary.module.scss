@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.qty {
    @include flexPosition($jusCon: space-between);
    padding: 10px 20px;
}
.msg {
    padding: 20px;
}
.btn {
    width: calc(100% - 40px);
    margin: 0 20px;
}
