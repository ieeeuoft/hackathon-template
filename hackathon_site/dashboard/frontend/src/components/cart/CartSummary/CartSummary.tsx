import React from "react";
import styles from "./CartSummary.module.scss";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TitledPaper from "components/general/TitledPaper/TitledPaper";
import { useDispatch, useSelector } from "react-redux";
import {
    cartTotalSelector,
    isLoadingSelector,
    submitOrder,
} from "slices/hardware/cartSlice";
import CircularProgress from "@material-ui/core/CircularProgress";
import { teamSizeSelector } from "slices/event/teamSlice";
import { isTestUserSelector } from "slices/users/userSlice";
import {
    hardwareSignOutEndDate,
    hardwareSignOutStartDate,
    maxTeamSize,
    minTeamSize,
} from "constants.js";

const CartSummary = () => {
    const isTestUser = useSelector(isTestUserSelector);
    const cartQuantity = useSelector(cartTotalSelector);
    const cartOrderLoading = useSelector(isLoadingSelector);
    const teamSize = useSelector(teamSizeSelector);
    const teamSizeValid = teamSize >= minTeamSize && teamSize <= maxTeamSize;
    const dispatch = useDispatch();
    const onSubmit = () => {
        if (cartQuantity > 0) {
            dispatch(submitOrder());
        }
    };
    const currentDateTime = new Date();
    const isOutsideSignOutPeriod =
        currentDateTime < hardwareSignOutStartDate ||
        currentDateTime > hardwareSignOutEndDate;

    return (
        <TitledPaper title="Cart Summary">
            <Container className={styles.qty}>
                <Typography variant="body2">Quantity</Typography>
                <Typography variant="body2" data-testid="cart-quantity-total">
                    {cartQuantity}
                </Typography>
            </Container>
            <Typography variant="body2" className={styles.msg}>
                Your entire team's order is here. Ensure that you have the correct
                number of hardware items before submitting.
            </Typography>
            <Button
                color="primary"
                variant="contained"
                className={styles.btn}
                disabled={
                    cartQuantity === 0 ||
                    cartOrderLoading ||
                    !teamSizeValid ||
                    (!isTestUser && isOutsideSignOutPeriod)
                }
                onClick={onSubmit}
                disableElevation
                data-testid="submit-order-button"
            >
                {cartOrderLoading ? (
                    <CircularProgress size={20} data-testid="order-loading-icon" />
                ) : (
                    "Submit order"
                )}
            </Button>
        </TitledPaper>
    );
};

export default CartSummary;
