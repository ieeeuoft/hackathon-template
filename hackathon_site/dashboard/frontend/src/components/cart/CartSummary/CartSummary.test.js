import React from "react";
import { render } from "@testing-library/react";
import CartSummary from "./CartSummary";
import { cartQuantity } from "testing/mockData";

it(`Renders correctly when it reads the number ${cartQuantity}`, () => {
    const { getByText } = render(<CartSummary cartQuantity={cartQuantity} />);
    expect(getByText(cartQuantity.toString())).toBeInTheDocument();
});
