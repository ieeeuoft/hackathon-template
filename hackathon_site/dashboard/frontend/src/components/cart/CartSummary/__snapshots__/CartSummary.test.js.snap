// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when it reads the number 0 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root title MuiTypography-h2 MuiTypography-noWrap"
  >
    Cart Summary
  </h2>
  <div
    class="MuiPaper-root paper MuiPaper-elevation3"
  >
    <div
      class="MuiContainer-root cardPaperQty MuiContainer-maxWidthLg"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Quantity
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        0
      </p>
    </div>
    <p
      class="MuiTypography-root cardPaperMsg MuiTypography-body2"
    >
      Your entire teamâ€™s order is here. Before you submit, refresh the page if someone added something new.
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Submit order
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
